### PR TITLE
Add support for numeric events

### PR DESCRIPTION
**Pull request by rorp**

### CODE REVIEW

**Comment by bennyhodl** [comment]
tACK

will have to update after https://github.com/benthecarman/kormir/pull/8

**Comment by benthecarman** [comment]
needs rebase now

File: kormir/src/lib.rs [hunk 1]
```diff
@@ -196,6 +195,161 @@ impl<S: Storage> Oracle<S> {
...
+                let sig = dlc::secp_utils::schnorrsig_sign_with_nonce(
+                    &self.secp,
+                    &msg,
+                    &self.key_pair,
+                    &nonce_key.secret_bytes(),
+                );
+                // verify our nonce is the same as the one in the announcement
+                debug_assert!(
+                    sig.encode()[..32] == nonce_key.x_only_public_key(&self.secp).0.serialize()
+                );
```
Comment (benthecarman) [inline kormir/src/lib.rs:329]: can you change the assert to be the same as this one

https://github.com/benthecarman/kormir/pull/11/files

File: kormir/src/lib.rs [hunk 2]
```diff
@@ -281,4 +435,88 @@ mod test {
 
         assert_eq!(rx, expected_nonce)
     }
+
```
Comment (benthecarman) [inline kormir/src/lib.rs:438]: can you add tests for a signed event too

**Review by benthecarman — COMMENTED** [review]
overall looks good

**Comment by benthecarman** [comment]
ah okay, can you fix the clippy issue then should be good to go

**Pull request merged**